csrc/optimization/alias_analysis.cpp

```diff
@@ -46,54 +88,44 @@ void findAliasesOfRoot(
   // to proactively make output aliases.
   // 2. It should handle more op types such as `Set.Permute`.
   // 3. It should detect alias between non-packed tensors.
-  std::queue<const TensorView*> q;
-  if (root->getMaybeAllocationDomain() == root->getMaybeRFactorDomain() &&
-      isContiguous(*root)) {
-    q.push(root);
-  }
-
-  while (!q.empty()) {
-    const TensorView* in_tv = q.front();
-    q.pop();
-
-    for (Expr* use : in_tv->uses()) {
-      if (!use->isA<ViewOp>()) {
-        continue;
-      }
+  if (ViewOp* view = dynamic_cast<ViewOp*>(expr)) {
+    TensorView* in = dynamic_cast<TensorView*>(view->in());
+    if (in == nullptr) {
```

Comment:
nitpick: are we expecting view op to have non-tensor IO? wondering if we should have assert or throw a warning here?

Reply:
Wdyt, @naoyam ? If the I/O is always a TensorView, I'd like to see that reflected in ViewOp's API. E.g. its `in()` and `out()` should return a TensorView instead of a Val and its constructor should do the assert.

Reply:
^^^yes, that's what I was asking as well.

Can `ViewOp` take a non-TensorView argument?

Reply:
In the case of `ViewOp`, I believe we can change the type to `TensorView`, but generally speaking, `TensorView` inputs and outputs are replaced with `kir::TensorIndex` during the lowering to the Kernel IR. For `ViewOp`, it's replaced with `LoadStoreOp` as part of fusion simplifications, so there should be no `ViewOp` with `kir::TensorIndex`.

Personally, I just don't remember which Expr types may have `TensorIndex` too, so I just tend to do `->as<TensorView>()` always or use something like `ir_utils::getTvOutput(Expr*)`.